### DISCUSSION TITLE
Is there a way to mask custom SPI parameters in the log message "Next time you run the server, just run: ..."?

### DISCUSSION THREAD

**Discussion by achimbitzer** [keycloak/keycloak#30042] [Q&A] [answered]
Hello,

I'm using a custom user storage provider with keycloak which requires some command line arguments, like the following example:

```
kc.sh start --spi-realm-restapi-extension-my-extension-someparam=abcd
```
On startup, keycloak prints this message:

```
Server configuration updated and persisted. Run the following command to review the configuration:

	kc.sh show-config

Next time you run the server, just run:

	kc.sh start --optimized --spi-realm-restapi-extension-my-extension-someparam=abcd
```

I would like to prevent printing the value of `--spi-realm-restapi-extension-my-extension-someparam` (`abcd` in this example).
Is it possible to mask this value in this log message?

I've already checked the keycloak source code and I think the proper way would be to register a custom `PropertyMapper` in `org.keycloak.quarkus.runtime.configuration.mappers.PropertyMappers` that returns `false` in `isMask()`. 
Unfortunately I couldn't find out how to add such a custom property mapper from my custom SPI. All instances of `PropertyMapper` seem to be hard coded in Keycloak. Did I miss something or is this actually impossible?

Any help is greatly apreciated, thank you in advance! 

Best regards, 
Achim

**Comment by shawkins** [comment, marked as answer]
> Did I miss something or is this actually impossible?

It is not currently possible if you provide the option via the cli. You can create an enhancement for that, but it will probably be low prioirty since the value is already being specified on the cli. You can alternatively include the option in your ENV or in the config file, then it won't be shown in the command.

**Reply by achimbitzer** [reply]
Thank you for your quick response
